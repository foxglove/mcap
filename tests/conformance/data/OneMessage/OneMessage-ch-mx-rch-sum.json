{
  "records": [
    {"type": "Header", "fields": [["library", ""], ["profile", ""]]},
    {
      "type": "Schema",
      "fields": [
        ["data", ["4", "5", "6"]],
        ["encoding", "c"],
        ["id", "1"],
        ["name", "Example"]
      ]
    },
    {
      "type": "Channel",
      "fields": [
        ["id", "1"],
        ["message_encoding", "a"],
        ["metadata", {"foo": "bar"}],
        ["schema_id", "1"],
        ["topic", "example"]
      ]
    },
    {
      "type": "Message",
      "fields": [
        ["channel_id", "1"],
        ["data", ["1", "2", "3"]],
        ["log_time", "2"],
        ["publish_time", "1"],
        ["sequence", "10"]
      ]
    },
    {"type": "DataEnd", "fields": [["data_section_crc", "1703938896"]]},
    {
      "type": "Channel",
      "fields": [
        ["id", "1"],
        ["message_encoding", "a"],
        ["metadata", {"foo": "bar"}],
        ["schema_id", "1"],
        ["topic", "example"]
      ]
    },
    {
      "type": "SummaryOffset",
      "fields": [
        ["group_length", "47"],
        ["group_opcode", "4"],
        ["group_start", "233"]
      ]
    },
    {
      "type": "Footer",
      "fields": [
        ["summary_crc", "277674795"],
        ["summary_offset_start", "280"],
        ["summary_start", "233"]
      ]
    }
  ],
  "meta": {"variant": {"features": ["ch", "mx", "rch", "sum"]}}
}
