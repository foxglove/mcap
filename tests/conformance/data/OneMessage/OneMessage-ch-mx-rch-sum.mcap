version https://git-lfs.github.com/spec/v1
oid sha256:e124eb30c25becdabf4ee4547245e55c7c4d7ca61779d72fae97f22cd0e8c0ac
size 343
