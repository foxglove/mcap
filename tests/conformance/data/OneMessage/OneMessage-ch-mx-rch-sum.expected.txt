Header library="" profile=""
ChannelInfo id=1 message_encoding=a metadata={foo=bar} schema=b schema_encoding=c schema_name=Example topic=example
Message channel_id=1 log_time=2 message_data=<010203> publish_time=1 sequence=10
MessageIndex channel_id=1 records={2=66}
ChannelInfo id=1 message_encoding=a metadata={foo=bar} schema=b schema_encoding=c schema_name=Example topic=example
ChunkIndex chunk_length=139 chunk_start_offset=25 compressed_size=100 compression="" end_time=2 message_index_length=31 message_index_offsets={1=166} start_time=2 uncompressed_size=100
SummaryOffset group_length=66 group_opcode=12 group_start=210
SummaryOffset group_length=0 group_opcode=12 group_start=276
SummaryOffset group_length=0 group_opcode=9 group_start=276
SummaryOffset group_length=87 group_opcode=7 group_start=276
Footer summary_offset_start=363 summary_start=210
